tests/marray/marray_custom_type.h

```diff
@@ -0,0 +1,158 @@
+#ifndef SYCL_CTS_TEST_MARRAY_MARRAY_CUSTOM_TYPE_H
+#define SYCL_CTS_TEST_MARRAY_MARRAY_CUSTOM_TYPE_H
+
+#include <cstdint>
+
+/// Custom type that meets the NumericType requirement.
+struct custom_int {
+  custom_int() = default;
+  constexpr custom_int(int i) : x{i} {};
+  operator int() { return x; }
+  custom_int(const custom_int& other) = default;
+  custom_int(custom_int&& other) noexcept = default;
+
+  custom_int& operator=(const custom_int& other) = default;
+
+  custom_int& operator=(custom_int&& other) noexcept = default;
+
+  custom_int& operator=(const int& val) {
+    *this = custom_int(val);
+    return *this;
+  }
+
+  custom_int& operator+=(const custom_int& val) {
+    x += val.x;
+    return *this;
+  }
+  custom_int& operator-=(const custom_int& val) {
+    x -= val.x;
+    return *this;
+  }
+  custom_int& operator*=(const custom_int& val) {
+    x *= val.x;
+    return *this;
+  }
+  custom_int& operator/=(const custom_int& val) {
+    x /= val.x;
+    return *this;
+  }
+  custom_int& operator%=(const custom_int& val) {
+    x %= val.x;
+    return *this;
+  }
+  custom_int& operator&=(const custom_int& val) {
+    x &= val.x;
+    return *this;
+  }
+  custom_int& operator|=(const custom_int& val) {
+    x |= val.x;
+    return *this;
+  }
+  custom_int& operator^=(const custom_int& val) {
+    x ^= val.x;
+    return *this;
+  }
+  custom_int& operator<<=(const custom_int& val) {
+    x <<= val.x;
+    return *this;
+  }
+  custom_int& operator>>=(const custom_int& val) {
+    x >>= val.x;
+    return *this;
+  }
+
+  custom_int operator+() const { return custom_int(+x); }
```

Comment:
It is not clear why you need all these casts
```suggestion
  custom_int operator+() const { return +x; }
```
?

Reply:
I've removed the casts.